openwisp_controller/pki/utils.py

```diff
@@ -0,0 +1,20 @@
+from django.core.exceptions import ValidationError
+
+
+class UnqiueCommonNameMixin(object):
+    def validate_unique(self, exclude=None):
+        super().validate_unique(exclude=exclude)
+        if (
+            self.organization is None
+            and self._meta.model.objects.filter(
+                organization=None, common_name=self.common_name
+            ).exists()
```

Comment:
I don't understand, isn't this redundant with `UniqueConstraint`?
and shouldn't  self.pk be excluded?

Reply:
> I don't understand, isn't this redundant with UniqueConstraint?

`UniqueConstraint` does not take into consideration unique NULL values due to the underlying implementation in the datbase enginer. 
Please refer  https://sqlite.org/faq.html#q26 

> shouldn't self.pk be excluded?

Yes, you are right.